src/GraphQL/DocumentType/PageSnippetType.php

```diff
@@ -50,5 +54,52 @@ public function build(&$config)
             'type' => Type::listOf($this->documentElementType),
             'resolve' => [$resolver, "resolveElements"]
         ];
+
+        $keyValue = new ListOfType(KeyValueType::getInstance());
+
+        $config['fields']['rendered'] = [
+            'type' => Type::string(),
+            'args' => [
+                'attributes' => [
+                    'type' => $keyValue,
+                    'description' => 'Attributes passed into the controller/action',
+                    'defaultValue' => []
+                ],
+                'query' => [
+                    'type' => $keyValue,
+                    'description' => 'Query Params passed into the controller/action',
+                    'defaultValue' => []
+                ],
+                'options' => [
+                    'type' => $keyValue,
+                    'description' => 'Options passed into the controller/action',
+                    'defaultValue' => []
+                ],
+                'use_layout' => [
+                    'type' => Type::boolean(),
+                    'description' => 'Disable Layout Rendering'
+                ]
+            ],
+            'resolve' => static function($value, $args) {
+                $documentId = $value['id'];
+                $document = PageSnippet::getById($documentId);
```

Comment:
do we need a permission check here ?

Reply:
just double checked, but if you don't have permission, this does not get loaded.

Reply:
True, it is already handled here:

https://github.com/pimcore/data-hub/blob/10a31a94c23a8c567d00a4a6b265fdf7139cff3a/src/GraphQL/Resolver/QueryType.php#L172

Thanks for checking